### ISSUE TITLE
BEP 55 hole punching

### ISSUE THREAD

**Issue by resession** [webtorrent/webtorrent#2213]
Haven't seen anything BEP 55(hole punching) for webtorrent. Any plans to support it in the future or is that out of scope for webtorrent?

**Comment by RangerMauve** [comment]
For reference, WebTorrent also supports raw UDP when running in node.js, and stuff like webtorrent-hybrid enable you to do run both UDP and WebRTC in paralell. (this is what's used in WebTorrent Desktop).

**Comment by paullouisageneau** [comment]
@RangerMauve Indeed, my statement may be misleading, BEP 55 can still be implemented for traditional Bittorrent connections over UDP when running out the browser.

**Comment by resession** [comment]
this issue was opened for running webtorrent on nodejs, since we have access to UDP on node, BEP 55 would make connections between peers a lot better.

**Comment by isc30** [comment]
Hey, did any of you implement the extension? I'm about to implement it in case no one else did 😃

**Comment by SilentBot1** [comment]
> Hey, did any of you implement the extension? I'm about to implement it in case no one else did 😃

Hey @isc30, UPNP/NAT-PMP has been implemented for TCP/uTP, and WebRTC is now in the default library as of WebTorrent >= 2.3.0 (which has hole punching via STUN), but BEP 55 hasn't been implemented for uTP connections, so feel free to give it a shot!
